azkaban-common/src/main/java/azkaban/server/SlaRequestUtils.java

```diff
@@ -85,8 +93,35 @@ public static List<SlaOption> parseSlaOptions(final HttpServletRequest req, fina
     return slaOptions;
   }
 
-  private static SlaOption parseSlaSetting(final String set, final String flowName, final List<String> emails,
-      final Map<String, Map<String, String>> alertersConfigs) throws ServletException {
+  private static Map<String, Map<String, List<String>>> getStringListMapFromStringMap(
+      final Map<String, Map<String, String>> alertersConfsFromReqParams) {
+    final Map<String, Map<String, List<String>>> result = new HashMap<>();
+    for (final String alerter : alertersConfsFromReqParams.keySet()) {
+      final Map<String, String> alerterConfigs = alertersConfsFromReqParams.get(alerter);
+      final Map<String, List<String>> processedConfigs = new HashMap<>();
+      for (final String propKey : alerterConfigs.keySet()) {
+        processedConfigs.put(propKey, getListFromString(alerterConfigs.get(propKey),
+            alerter.equals(Emailer.ALERTER_NAME)));
+      }
+      result.put(alerter, processedConfigs);
+    }
+    return result;
+  }
+
+  private static List<String> getListFromString(final String str, final boolean spacesAreDelims) {
+    if (str == null || str.trim().isEmpty()) {
+      return Collections.emptyList();
+    }
+    String delimRegex = Constants.ConfigurationKeys.ALERTER_PARAM_VALUE_DELIMITER;
+    if (spacesAreDelims) {
+      delimRegex = "\\s*,\\s*|\\s*;\\s*|\\s+";
```

Comment:
Can this be delimRegex += |\\s+ ?

Reply:
I thought that might make it harder to understand the regular expression therefore I kept it literal

Reply:
Makes sense!